activejob/lib/active_job/test_helper.rb

```diff
@@ -10,6 +10,7 @@ module TestHelper
 
     def before_setup # :nodoc:
       test_adapter = queue_adapter_for_test
+      self.class.to_s.gsub(/Test\z/, "").safe_constantize
```

Comment:
> What happens if users call tests SomethingUnconventional?

`ActiveJob::TestCase` does not have `tests` method. So, calling `tests` will an error.  

> but is this the best way we can find to solve the issue?  

I think your concern is correct. I think again about the solution.